external/fv3fit/fv3fit/_shared/novelty_detector.py

```diff
@@ -19,24 +19,24 @@ class NoveltyDetector(Predictor, abc.ABC):
 
     _NOVELTY_OUTPUT_VAR = "is_novelty"
     _SCORE_OUTPUT_VAR = "novelty_score"
+    _CENTERED_SCORE_OUTPUT_VAR = "centered_score"
 
     def __init__(self, input_variables: Iterable[Hashable]):
-        output_variables = [self._NOVELTY_OUTPUT_VAR, self._SCORE_OUTPUT_VAR]
+        output_variables = [
+            self._NOVELTY_OUTPUT_VAR,
+            self._SCORE_OUTPUT_VAR,
+            self._CENTERED_SCORE_OUTPUT_VAR,
+        ]
         super().__init__(input_variables, output_variables)
 
     def predict_novelties(
-        self, X: xr.Dataset, cutoff: Optional[float] = None
-    ) -> xr.Dataset:
-        if cutoff is None:
-            cutoff = self._get_default_cutoff()
+        self, X: xr.Dataset, cutoff: float = 0
+    ) -> Tuple[xr.DataArray, xr.Dataset]:
+        diagnostics = self.predict(X)
 
-        score_dataset = self.predict(X)
+        is_novelty = xr.where(
```

Comment:
Should this be moved to `OutOfSampleModel`?  It seems like`OutOfSampelModel` now owns the idea of cutoff and should keep it as private as possible. The cutoff is only used to compute the `is_novelty` output...which I think is only used for diagnostics. Since the `OutOfSampleModel` can compute this on its own, it's probably best to do so.

Or maybe just delete the code since it can be easily diagnosed from `centered_score` anyway.

tl;dr;. NoveltyDetector returns a centered score. OutOfSampleModel decides if it is a "novelty" or not based on it's own configured cutoff.

Reply:
That sounds good, I think it'd be cleaner to refactor and change the novelty detector's tests in a follow-up PR though and try keep this one mostly related to adding the taper function.

Reply:
Sgtm too. Once we do that, there probably isn't a reason for `predict_novelties` to exist as a function, and the only point of `NoveltyDetector` would be to specify the output variables of any `Predictor` implementing it.